### PR TITLE
libseccomp (new formula)

### PR DESCRIPTION
**Pull request by iMichka**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: .github/workflows/tests-linux.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
+name: Linux CI
```
Comment (MikeMcQuaid) [inline .github/workflows/tests-linux.yml:1]: I don't think we're ready to be adding Linux-only CI to Homebrew/homebrew-core yet.

File: Formula/libseccomp.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  bottle do
+    cellar :any_skip_relocation
+    sha256 "ab6d430d4c758ce3cc55e5ec6ba8609196655bc03988271c25ad086ff54f823e" => :x86_64_linux
+  end
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+  depends_on :linux
```
Comment (MikeMcQuaid) [inline Formula/libseccomp.rb:15]: Don't think we're ready for these yet.

**Review by MikeMcQuaid — DISMISSED** [review]

**Comment by iMichka** [comment]
I did not give not enough context, sorry:
-  We have around 5-10 formulae that are linux only, and some of them are linux-only dependencies which I need
- I would like to slowly add these to homebrew-core, so I can add these dependencies, thus reducing the diff
- I need to start somewhere, and reducing the diff as much as possible gives me more visibility on what remains to be done

I added:
```
paths:
   - "Formula/libseccomp.rb"
```

to the linux CI file. I will extend that list slowly for these 5-10 linux-only formulae, so that these and only these formulae get tested. I do not plan to enable the linux-only CI for other formulae now.

This will not impact homebrew-core: it will not impact maintainers or our users. There will be no bottle upload here, just CI testing (as this is the condition to get a new formula added here). The final bottle upload will be done in linuxbrew-core.

These formulae will remain untouched in homebrew-core for the next 6 months without much usage, until we finish the migration. Version updates will need to be done here first though.

This will also allow to prepare the linux CI here and test things out over the next months.

**Comment by iMichka** [comment]
I need to split this PR in 2 separate commits and fix the CI: once the build is green, the plan is to open a separate PR with the CI setup alone.

**Comment by MikeMcQuaid** [comment]
TBH: I think it's OK to migrate these for now and rely entirely on `tap_syntax` (which is running on Linux and will check these pass `brew style`, `brew readall` and `brew audit`.



> paths:

This isn't a bad idea but I'd want to have a way of verifying that e.g.:
- all these files exist
- this correctly lists all linux-only dependencies

and similar. That could perhaps be a simple bash script check run by the `tap_syntax` job?

**Comment by MikeMcQuaid** [comment]
https://github.com/Homebrew/homebrew-core/pull/59372/checks?check_run_id=964134842#step:5:38

**Comment by iMichka** [comment]
I can write something which fetches the list of `depends_on :linux` formulae in the `tap_syntax` workflow, and check that they are indeed tested by the linux-only CI (by checking the list of paths that are being tested). We need to have the full linux CI (not only  `tap_syntax`), because someone might want to version bump one of these formulae one day, and we need to make sure it properly builds.

**Comment by MikeMcQuaid** [comment]
> I can write something which fetches the list of `depends_on :linux` formulae in the `tap_syntax` workflow, and check that they are indeed tested by the linux-only CI (by checking the list of paths that are being tested).

Yeh, sounds good.



> We need to have the full linux CI (not only `tap_syntax`), because someone might want to version bump one of these formulae one day, and we need to make sure it properly builds.

We'd also need to fix `brew test-bot` to handle this correctly. Right now I think just adjusting the messaging to state that the PR shouldn't be merged may be enough 😁.

**Comment by iMichka** [comment]
I could add a flag similar to `--skip-recursive-dependents` to test bot, so that only the current formula is tested. What do you think?

File: .github/workflows/tests-linux.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            rm -rvf bottles/home
+
+        - name: Count bottles
+          id: bottles
+          if: always()
+          run: |
+            cd bottles
+            count=$(ls *.json | wc -l | xargs echo -n)
+            echo "$count bottles"
+            echo "::set-output name=count::$count"
```
Comment (MikeMcQuaid) [inline .github/workflows/tests-linux.yml:58]:
```suggestion
```

None of this is needed if we're not using the bottles.

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -30,6 +30,17 @@ jobs:
 
     - run: brew test-bot --only-tap-syntax
 
+    - name: Check linux-only formulae coverage
+      run: |
+        FILES=$(grep .github/workflows/tests-linux.yml -e .rb | sed -e 's/-//g' | sed -e 's/ //g' | sed -e 's/"//g')
```
Comment (MikeMcQuaid) [inline .github/workflows/tests.yml:35]:
```suggestion
        FILES=$(grep -oE 'Formula/[a-z0-9_\.\-]+' .github/workflows/tests-linux.yml)
```

File: .github/workflows/tests.yml [hunk 5]
```diff
@@ -30,6 +30,17 @@ jobs:
...
+        for FILE in $FILES; do
+          if [ ! -f "$FILE" ]; then
+            echo "$FILE does not exist."
```
Comment (MikeMcQuaid) [inline .github/workflows/tests.yml:38]: Can this error message be improved?

File: .github/workflows/tests.yml [hunk 6]
```diff
@@ -30,6 +30,17 @@ jobs:
...
+            exit 1
+          fi
+          grep -q "depends_on :linux" "$FILE" || echo "$FILE is not a linux-only formula."; exit 1
```
Comment (MikeMcQuaid) [inline .github/workflows/tests.yml:41]: Can this also use an `if`

File: .github/workflows/tests.yml [hunk 7]
```diff
@@ -30,6 +30,17 @@ jobs:
 
     - run: brew test-bot --only-tap-syntax
 
+    - name: Check linux-only formulae coverage
```
Comment (MikeMcQuaid) [inline .github/workflows/tests.yml:33]:
```suggestion
    - name: Check .github/workflows/tests-linux.yml formulae coverage
```

File: Formula/libseccomp.rb [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}"
+    system "make", "install"
+  end
+
+  test do
+    ver = version.to_s.split(".")
+    ver_major = ver[0]
+    ver_minor = ver[1]
```
Comment (MikeMcQuaid) [inline Formula/libseccomp.rb:24]:
```suggestion
    ver_major, ver_minor, = version.to_s.split(".")
```

**Comment by iMichka** [comment]
```
2020-08-12T21:40:25.5210177Z [31m==>[0m [1m[31mFAILED[0m[0m
2020-08-12T21:40:25.5211073Z [1mPlease note that these warnings are just used to help the Homebrew maintainers
2020-08-12T21:40:25.5211500Z with debugging if you file an issue. If everything you use Homebrew for is
2020-08-12T21:40:25.5212001Z working fine: please don't worry or file an issue; just ignore this. Thanks![0m
2020-08-12T21:40:25.5212086Z 
2020-08-12T21:40:25.5212450Z [33mWarning:[0m Suspicious https://github.com/Homebrew/linuxbrew-core git origin remote found.
2020-08-12T21:40:25.5212581Z The current git origin is:
2020-08-12T21:40:25.5212816Z   https://github.com/Homebrew/homebrew-core
2020-08-12T21:40:25.5212890Z 
2020-08-12T21:40:25.5213107Z With a non-standard origin, Homebrew won't update properly.
2020-08-12T21:40:25.5213213Z You can solve this by setting the origin remote:
2020-08-12T21:40:25.5213718Z   git -C "/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core" remote set-url origin [4mhttps://github.com/Homebrew/linuxbrew-core[24m
2020-08-12T21:40:25.5213845Z 
2020-08-12T21:40:25.5214117Z [33mWarning:[0m You have uncommitted modifications to Homebrew/homebrew-core.
2020-08-12T21:40:25.5214232Z If this is a surprise to you, then you should stash these modifications.
2020-08-12T21:40:25.5214344Z Stashing returns Homebrew to a pristine state but can be undone
2020-08-12T21:40:25.5214448Z should you later need to do so for some reason.
2020-08-12T21:40:25.5214704Z   cd /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core && git stash && git clean -d -f
2020-08-12T21:40:25.5214792Z 
2020-08-12T21:40:25.5214880Z Uncommitted files:
2020-08-12T21:40:25.5214970Z   ?? steps_output.txt
2020-08-12T21:40:25.5215014Z 
2020-08-12T21:40:25.5215101Z Error: 1 failed steps!
2020-08-12T21:40:25.5215190Z brew doctor
2020-08-12T21:40:25.5533268Z ##[error]Process completed with exit code 1.
2020-08-12T21:40:25.5577348Z ##[group]Run cat bottles/steps_output.txt
```

**Comment by iMichka** [comment]
Needs https://github.com/Homebrew/actions/pull/57

**Comment by iMichka** [comment]
@dawidd6 fixed the workflow. This is good for a final review.

**Review by SMillerDev — DISMISSED** [review]

File: .github/workflows/tests-linux.yml [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+      - name: Set up Homebrew
+        uses: Homebrew/actions/setup-homebrew@master
+
+      - run: brew test-bot --only-cleanup-before
+
+      - name: Run brew test-bot --only-formulae
+        run: |
+          mkdir ~/bottles
+          cd ~/bottles
+          brew test-bot --only-formulae
```
Comment (MikeMcQuaid) [inline .github/workflows/tests-linux.yml:24]:
```suggestion
        run: brew test-bot --only-formulae
```

as we're not using the bottles.

**Review by MikeMcQuaid — COMMENTED** [review]
One minor tweak and then ✅ from me.

File: .github/workflows/tests-linux.yml [hunk 10]
```diff
@@ -0,0 +1,24 @@
+name: Linux CI
```
Comment (MikeMcQuaid) [inline .github/workflows/tests-linux.yml:1]:
```suggestion
name: GitHub Actions CI
```

File: .github/workflows/tests-linux.yml [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+on:
+  pull_request:
+    paths:
+      - "Formula/libseccomp.rb"
+jobs:
+  tests:
```
Comment (MikeMcQuaid) [inline .github/workflows/tests-linux.yml:7]:
```suggestion
  tests_linux:
```

**Review by MikeMcQuaid — COMMENTED** [review]
One more thought with naming.

**Comment by iMichka** [comment]
@MikeMcQuaid You asked me to move the build out of the bottles folder. I need to create a new folder and move there, because else I am getting the following error:
```
==> brew uninstall --force autoconf automake expat libtool m4 perl xz
Error: Invalid cross-device link @ rb_file_s_link - (./libseccomp--2.4.2.x86_64_linux.bottle.tar.gz, /github/home/.cache/Homebrew/downloads/a9057ed1a0c9d458f2837f27e0406624e5368665aa70fcd001517b767d0a88f2--libseccomp-2.4.2.x86_64_linux.bottle.tar.gz)
##[error]Process completed with exit code 1.
```

This is because the working directory is a mounted folder between the host and the docker container.

**Review by dawidd6 — APPROVED** [review]

**Review by MikeMcQuaid — COMMENTED** [review]
Looks good to me!

**Pull request merged**